contracts/modules/common/BaseFeature.sol

```diff
@@ -0,0 +1,173 @@
+     * @notice Throws if the sender is not the owner of the target wallet or the feature itself.
+     */
+    modifier onlyWalletOwnerOrFeature(address _wallet) {
+        // Wrapping in an internal method reduces deployment cost by avoiding duplication of inlined code
+        verifyOwnerOrAuthorisedFeature(_wallet, msg.sender);
+        _;
+    }
+
+    constructor(
+        ILockStorage _lockStorage,
+        IVersionManager _versionManager,
+        bytes32 _name
+    ) public {
+        lockStorage = _lockStorage;
+        versionManager = _versionManager;
+        emit FeatureCreated(_name);
+    }
+
+    /**
+    * @inheritdoc IFeature
+    */
+    function recoverToken(address _token) external virtual override {
+        uint total = ERC20(_token).balanceOf(address(this));
+        _token.call(abi.encodeWithSelector(ERC20(_token).transfer.selector, address(versionManager), total));
+    }
+
+    /**
+     * @notice Inits the feature for a wallet by logging an event.
+     * The method can only be called by the VersionManager.
+     * @param _wallet The wallet.
+     */
+    function init(address _wallet) public virtual override onlyVersionManager {
+        emit FeatureInitialised(_wallet);
+    }
+
+    /**
+     * @inheritdoc IFeature
+     */
+    function getRequiredSignatures(address, bytes calldata) external virtual view override returns (uint256, OwnerSignature) {
+        revert("BF: disabled method");
+    }
+
+    /**
+     * @inheritdoc IFeature
+     */
+    function getStaticCallSignatures() external virtual override view returns (bytes4[] memory _sigs) {}
+
+    /**
+     * @inheritdoc IFeature
+     */
+    function isFeatureAuthorisedInVersionManager(address _wallet, address _feature) public override view returns (bool) {
+        return versionManager.isFeatureAuthorised(_wallet, _feature);
+    }
+    
+     /**
+     * @notice Helper method to check if an address is the owner of a target wallet.
+     * @param _wallet The target wallet.
+     * @param _addr The address.
+     */
+    function isOwner(address _wallet, address _addr) internal view returns (bool) {
+        return IWallet(_wallet).owner() == _addr;
+    }
+
+    /**
+     * @notice Verify that the caller is an authorised feature or the wallet owner.
+     * @param _wallet The target wallet.
+     * @param _sender The caller.
+     */
+    function verifyOwnerOrAuthorisedFeature(address _wallet, address _sender) internal view {
+        require(isFeatureAuthorisedInVersionManager(_wallet, _sender) || isOwner(_wallet, _sender), "BF: must be owner or feature");
+    }
+
+    /**
+     * @notice Helper method to invoke a wallet.
+     * @param _wallet The target wallet.
+     * @param _to The target address for the transaction.
+     * @param _value The value of the transaction.
+     * @param _data The data of the transaction.
+     */
+    function checkAuthorisedFeatureAndInvokeWallet(address _wallet, address _to, uint256 _value, bytes memory _data)
```

Comment:
Why did we change the name from `invokeWallet` to `checkAuthorisedFeatureAndInvokeWallet`? It adds a lot of renaming in all the features and the method is just invoking the wallet.

Reply:
To avoid a name clash with  `VersionManager.invokeWallet` which is also a feature. I could rename `BaseFeature.checkAuthorisedFeatureAndInvokeWallet` back to `BaseFeature.invokeWallet` and rename `VersionManager.invokeWallet` to something else (e.g. `VersionManager.checkAuthorisedFeatureAndInvokeWallet` unless you can think of a better name)